src/commons/application/reducers/SessionsReducer.ts

```diff
@@ -88,6 +91,23 @@ export const SessionsReducer: Reducer<SessionState> = (
         ...state,
         gameState: action.payload
       };
+    case ADD_KEYSTROKE_LOG:
+      const logsCopy = state.keystrokeLogs;
```

Comment:
Use [...state.keystrokeLogs] to ensure that the object is copied instead of its reference being copied.